pkg/odo/cli/component/list.go

```diff
@@ -47,21 +56,45 @@ func NewListOptions() *ListOptions {
 // Complete completes log args
 func (lo *ListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 
-	if util.CheckKubeConfigExist() {
-		klog.V(4).Infof("New Context")
-		lo.Context = genericclioptions.NewContext(cmd)
+	if experimental.IsExperimentalModeEnabled() {
+		// Add a disclaimer that we are in *experimental mode*
+		log.Experimental("Experimental mode is enabled, use at your own risk")
+
+		lo.Context = genericclioptions.NewDevfileContext(cmd)
+
 	} else {
-		klog.V(4).Infof("New Config Context")
-		lo.Context = genericclioptions.NewConfigContext(cmd)
+		if util.CheckKubeConfigExist() {
+			klog.V(4).Infof("New Context")
+			lo.Context = genericclioptions.NewContext(cmd)
+		} else {
+			klog.V(4).Infof("New Config Context")
+			lo.Context = genericclioptions.NewConfigContext(cmd)
+
+		}
+	}
+
+	lo.Client = genericclioptions.Client(cmd)
+	dcSupport, err := lo.Client.IsDeploymentConfigSupported()
 
+	if err != nil {
```

Comment:
i think it is safe to assume `dcSupport=false` here rather than returning error, because on a kubernetes cluster it might not return `404`.  Not very sure on this, but faced this problem in one of my pr and @mik-dass also ignored it here
https://github.com/openshift/odo/blob/master/pkg/component/component.go#L577

Reply:
DC is not present on kubernetes for sure.

Reply:
yeah, fair point though, this check shouldn't fail `odo`. and error would mean we just say `dcSupport` is not there.

Reply:
Yup DC is not supported on k8s for sure, but using this (we cannot avoid this)`const ClusterVersionGroup = "apps.openshift.io"` for discovery client on kubernetes cluster, we wont't get `kerrors.IsNotFound(err)`, so if we return error here, it will always on k8s cluster. So IMO it is safe to assume `false` or  check for that error too here 
https://github.com/openshift/odo/pull/3134/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR3287